src/main/java/luke/Luke.java

```diff
@@ -0,0 +1,301 @@
+            printReply(String.format("Task added: %s\n  %s",
+                    tasks.get(tasks.size() - 1).toString(), numberOfTasksMessage()));
+            break;
+        case DEADLINE:
+            for (int i = 0; i < args.length; i++) {
+                if (args[i].startsWith("/by")) {
+                    idx = i;
+                }
+            }
+            if (idx == -1) {
+                throw new InsufficientArguments("Deadline needs to be specified");
+            }
+            description = String.join(" ", Arrays.copyOf(args, idx));
+            deadlineStr = String.join(" ", Arrays.copyOfRange(args, idx + 1, args.length));
+            tasks.add(new Deadline(description, deadlineStr));
+            printReply(String.format("Added deadline: %s\n  %s",
+                    tasks.get(tasks.size() - 1).toString(), numberOfTasksMessage()));
+            break;
+
+        case EVENT:
+            for (int i = 0; i < args.length; i++) {
+                if (args[i].startsWith("/from")) {
+                    fromIdx = i;
+                } else if (args[i].startsWith("/to")) {
+                    toIdx = i;
+                }
+            }
+            if (fromIdx == -1) {
+                throw new InsufficientArguments("From when???");
+            }
+            if (toIdx == -1) {
+                throw new InsufficientArguments("To when???");
+            }
+            description = String.join(" ", Arrays.copyOf(args, fromIdx));
+            fromStr = String.join(" ", Arrays.copyOfRange(args, fromIdx + 1, toIdx));
+            toStr = String.join(" ", Arrays.copyOfRange(args, toIdx + 1, args.length));
+            tasks.add(new Event(description, fromStr, toStr));
+            printReply(String.format("Added event: %s\n %s",
+                    tasks.get(tasks.size() - 1).toString(), numberOfTasksMessage()));
+            break;
+        case DELETE:
+            if (args.length == 0) {
+                throw new InsufficientArguments("Delete command needs an index");
+            }
+            try {
+                idx = Integer.parseInt(args[0]) - 1;
+            } catch (NumberFormatException e) {
+                throw new IncorrectInput("Please input an integer");
+            }
+            if (idx < 0 || idx >= tasks.size()) {
+                throw new IncorrectInput("Invalid index");
+            }
+            Task taskToDelete = tasks.get(idx);
+            tasks.remove(taskToDelete);
+            printReply(String.format("Removed task:\n  %s\nNow you have %d %s in the list.",
+                    taskToDelete.toString(), tasks.size(), tasks.size() > 1 ? "tasks" : "task"));
+            break;
+        default:
+            throw new InvalidCommand("Invalid command");
+        }
+    }
+
+    private static void sendMessage(String userInput) {
+        String[] inputArr = userInput.split(" ");
+        String command = inputArr[0];
+
+        if (command.equalsIgnoreCase("bye")) {
```

Comment:
This is a very long piece of code. You may want to explore inheritance together with polymorphism to cut down the amount of code here.